robotframework-ls/tests/robotframework_ls_tests/completions/test_variables_completions.py

```diff
@@ -375,3 +375,45 @@ def test_variables_completions_in_variable_files_yaml(
         CompletionContext(doc, workspace=workspace.ws)
     )
     data_regression.check(completions)
+
+
+def test_dictionary_variables_completions_with_dollar(
+    workspace, libspec_manager, data_regression
+):
+    from robotframework_ls.impl.completion_context import CompletionContext
+    from robotframework_ls.impl import variable_completions
+
+    workspace.set_root("case10", libspec_manager=libspec_manager)
+    doc = workspace.get_doc("case10.robot")
+    doc.source = """*** Variables ***
+&{ROBOT}   Name=Robot Framework   Version=4.0
+
+***Test Cases***
+Test dictionary variable completion
+   Log to Console   ${ROB"""
+
+    completions = variable_completions.complete(
+        CompletionContext(doc, workspace=workspace.ws)
+    )
+    data_regression.check(completions)
```

Comment:
Note: the files generated by the data_regression must be committed.

Reply:
How do I generate them?

Reply:
When you run the test it should automatically generate the data regression files (it's a `.yaml` which will be placed in a `test_variables_completions` folder next to the `test_variables_completions.py`).

Then, after re-running the test, if the results are different, it will fail (to regenerate you can just delete the related file and rerun -- if it's not there the file will be generated and the test will fail).

Reply:
Note that if you have full paths/uris related to your system, the test will fail due to such differences (so, it's possible that you have to change `completions` a bit to fix such references before sending it to the `data_regression.check`).